pkgs/tools/archivers/peazip/default.nix

```diff
@@ -0,0 +1,95 @@
+{ mkDerivation
+, lib
+, fetchFromGitHub
+, wrapQtAppsHook
+, fpc
+, lazarus
+, xorg
+, libqt5pas
+, _7zz
+, archiver
+, brotli
+, upx
+, zpaq
+, zstd
+}:
+
+mkDerivation rec {
+  pname = "peazip";
+  version = "8.6.0";
+
+  src = fetchFromGitHub {
+    owner = "peazip";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-MF0MwIKV/25lm+IKRdFXSMiXSQxTKksGBj/r0td+v/o=";
+  };
+  sourceRoot = "source/peazip-sources";
+
+  nativeBuildInputs = [
+    wrapQtAppsHook
+    lazarus
+    fpc
+  ];
+
+  buildInputs = [
+    xorg.libX11
+    libqt5pas
+  ];
+
+  NIX_LDFLAGS = "--as-needed -rpath ${lib.makeLibraryPath buildInputs}";
+
+  buildPhase = ''
+    HOME=$(mktemp -d) lazbuild --lazarusdir=${lazarus}/share/lazarus --widgetset=qt5 --build-all project_pea.lpi && [ -f pea ]
+    HOME=$(mktemp -d) lazbuild --lazarusdir=${lazarus}/share/lazarus --widgetset=qt5 --build-all project_peach.lpi && [ -f peazip ]
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # Binaries
+    install -Dm755 peazip $out/bin/peazip
+    install -Dm755 pea $out/bin/pea
+
+    ## Symlink the available compression algorithm programs.
+    mkdir -p $out/bin/res/bin/7z
+    ln -s ${_7zz}/bin/7zz $out/bin/res/bin/7z/7z
```

Comment:
The problem is that peazip is looking for the executables in exactly those directory relative to it. I haven't found any way to change the paths in the source code either. Can we put that folder somewhere else and then symlink it to `$out/bin/res`?

Reply:
Pascal is a very basic language. If we find the path that references this we can just change it to something sensible.

> Can we put that folder somewhere else and then symlink it to `$out/bin/res`?

No, we have the source code so we can properly fix this. What is the error when the binary is missing? Can we search for that in the code?

Reply:
> What is the error when the binary is missing?

I used to get an error before the window even appeared but I don't get that error anymore. There's only an error if you try to do something in the application.

Reply:
![peazip screenshot when trying to extract without required executables](https://0x0.st/oArv.png
)

Reply:
`rg res/bin` doesn't bring up any sort of `7z="res/bin/7z/7z"`, only readmes.

Reply:
I don't think we should package it if we can't find the source of this error message.

Reply:
> We are not putting something under $out/bin/res. We need to find another directory.

Is there a reason for this 🤔 ? I don't think this a huge issue, it should be a pure approach anyway.

Maybe we will have a `res` PATH exposed in `PATH` (since directories are executables), but again, this shouldn't be a huge issue.

Reply:
> I don't think this a huge issue

It is. Only executable binaries belong into ``$out/bin``. Not directories.

Reply:
Yeah, I know. But do this generates a huge problem, or it is more of a philosophical problem?